students/carlos_novoa/lesson07/html_render.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+
+"""
+Lesson7 - HTML Renderer
+"""
+
+
+class Element:
+    'Generates HTML tags, content, and attributes'
+
+    version = '0.1'
+    indent = '  '
+    tag = ''
+
+    def __init__(self, content=None, **kwargs):
+        self.content = [] if content is None else [content]
+        self.atts = kwargs if kwargs else ''
+        self.tag = self.tag if self.tag else 'html'
+
+    @staticmethod
+    def format_atts(atts=None):
+        if atts:
+            s = ''.join(' {}="{}"'.format(k, v) for k, v in atts.items())
+        else:
+            s = ''
+        return s
+
+    def append(self, content):
+        self.content.append(content)
+
+    def render(self, file_out, cur_ind=""):
+        atts = self.format_atts(self.atts)
+        file_out.write("{}<{}{}>\n".format(cur_ind, self.tag, atts))
+        for node in self.content:
+            if isinstance(node, Element):
+                node.render(file_out, cur_ind + self.indent)
+            else:
+                if self.tag == 'a':
+                    file_out.write('{}{}\n'.format(cur_ind, node))
+                else:
+                    file_out.write(self.indent)
+                    file_out.write('{}{}\n'.format(cur_ind, node))
+        file_out.write("{}</{}>\n".format(cur_ind, self.tag))
+
+
+class Html(Element):
+    def __init__(self, content=None, **kwargs):
+        self.tag = 'html'
```

Comment:
`tag` should be a class attribute (keep it how it was initially defined at class level on `Element`, overriding the value for each subclass) , since it will be shared by all instances of the `Html` class